test/e2e/e2e_test.go

```diff
@@ -153,3 +256,190 @@ var _ = Describe("e2e control plane", func() {
+		}
+		haMode = false
+		// Attempt to retrieve the pod name that will run the external interface for e2e control-plane ha mode
+		if kubectlOut == "''" {
+			haMode = true
+			kubectlOut, err = framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorHaFlag)
+			if err != nil {
+				framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnHaWorkerNode, err)
+			}
+		}
+		// Fail the test if no pod is matched within the specified node
+		if kubectlOut == "''" {
+			framework.Failf("Unable to locate container %s on any known nodes", ovnContainer)
+		}
+	})
+
+	It("Should validate connectivity within a hybrid-sdn namespace of pods on separate worker nodes", func() {
+		var err error
+		var pingTarget string
+		var ciWorkerNodeSrc string
+		var ciWorkerNodeDst string
+		dstPingPodName := "e2e-dst-ping-pod"
+		getPodIPRetry := 15
+		command := []string{"bash", "-c", "sleep 20000"}
+		// HA CI mode runs a named set of nodes with a prefix of ovn-control-plane
+		if haMode {
+			By(fmt.Sprintf("Creating a container on node %s and verifying connectivity to a pod on node %s", ovnHaWorkerNode2, ovnHaWorkerNode))
+			framework.Logf("Detected a HA mode KIND environment")
+			ciWorkerNodeSrc = ovnHaWorkerNode
+			ciWorkerNodeDst = ovnHaWorkerNode2
+		}
+		// non-HA CI mode runs a named set of nodes with a prefix of ovn-worker
+		if !haMode {
+			By(fmt.Sprintf("Creating a container on node %s and verifying connectivity to a pod on node %s", ovnWorkerNode2, ovnWorkerNode))
+			ciWorkerNodeSrc = ovnWorkerNode
+			ciWorkerNodeDst = ovnWorkerNode2
+		}
+		// Create the pod that will be used as the destination for the connectivity test
+		createGenericPod(f, dstPingPodName, ciWorkerNodeDst, command)
+		// There is a condition somewhere with e2e WaitForPodNotPending that returns ready
+		// before calling for the IP address will succeed. This simply adds some retries.
+		for i := 1; i < getPodIPRetry; i++ {
+			pingTarget, err = getPodAddress(dstPingPodName, f.Namespace.Name)
+			if err != nil {
+				framework.Logf("Warning unable to query the test pod on node %s %v", ciWorkerNodeSrc, err)
+			}
+			if pingTarget != "<no value>" {
+				framework.Logf("Destination ping target for %s is %s", dstPingPodName, pingTarget)
+				break
+			}
+			time.Sleep(time.Second * 3)
+			framework.Logf("Retry attempt %d to get pod IP from initializing pod %s", i, dstPingPodName)
+		}
+		// Fail the test if no address is ever retrieved
+		if pingTarget == "<no value>" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, test will fail", dstPingPodName)
+		}
+		// Fail the test if the destination IP is empty
+		if pingTarget == "" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, failing test", dstPingPodName)
+		}
+		// Spin up another pod that attempts to reach the previously started pod on separate nodes
```

Comment:
After we verify connectivity, we should verify that no traffic crossed the br-ext vxlan tunnel if it exists, right?

Reply:
Done, great idea, ty.